pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -1270,6 +1282,30 @@ cdef _broadcast_floordiv_td64(int64_t value, object other,
         return res
 
 
+cdef inline _require_td64_has_unit(object value):
+    """
+    Require that a timedelta64 scalar have a unit specified,
+    i.e. 'timedelta64[ns]' is OK but 'timedelta64' is not.
+
+    Parameters
+    ----------
+    value : object
+        assumed to have a `dtype` attribute
+
+    Raises
+    ------
+    ValueError
+    """
+    if value.dtype == 'm8':
+        # i.e. has no unit
```

Comment:
pls make this more concise (e.g. a single if)

Reply:
The exception message could be shortened, but everything else is needed.  Combining the two ifs into one line would hurt clarity.

Reply:
you can just use ``missing.is_null_datetimelike`` instead of repeating code here. pls use a single if.